src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -563,9 +564,10 @@ private void WriteManifest(ZipArchive package, int minimumManifestVersion, strin
             }
         }
 
-        private HashSet<string> WriteFiles(ZipArchive package)
+        private HashSet<string> WriteFiles(ZipArchive package, out HashSet<string> filesWithoutExtensions)
```

Comment:
you shouldn't need to use out here, the hashset is passed by value so it will be modified no matter what